src/pages/api/pusher/channels/[course].ts

```diff
@@ -30,7 +32,14 @@ export default async function handler(
   if (channelData.status === 200) {
     const body = await channelData.json()
     const channelInfo = body.channels
-    res.json(channelInfo)
+
+    const contact_ids = keys(channelInfo).map((channel) => {
+      return last(channel.split('@@')) || ''
```

Comment:
the `@@` is a separator used in naming the private channels on Pusher where the second half of it is the contact guid for the learner

they "generally suggest something shaped like this" in this article

https://support.pusher.com/hc/en-us/articles/360019620253-How-can-I-implement-large-presence-channels-on-Channels-

they have a more straightforward presence feature but it's limited to 100 connections... maybe it's just wishful thinking on my part ;P